helm/hpcc/templates/_warnings.tpl

```diff
@@ -52,4 +52,79 @@ Pass in dict with root and warnings
   {{- $_ := set $warning "msg" (printf "The configuration contains ephemeral planes: %s" (print (uniq $match.ephemeral)) )  -}}
   {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
  {{- end -}}
+ {{- /* Warn if resources not defined for any cost component or default cpuRate being used */ -}}
+ {{- if eq .root.Values.global.cost.perCpu 0.0565000000001 -}}
+  {{- $_ := set $ctx "usingDefaultCpuCost" "true" -}}
+ {{- end -}}
+ {{- $_ := set $ctx "missingResources" list -}}
+ {{- $_ := set $ctx "defaultCpuRateComponents" list -}}
+ {{- range $component := .root.Values.eclagent -}}
+  {{- if and (not $component.disabled) (eq $component.type "hthor" ) -}}
+   {{- $hasResources := include "hpcc.checkResources" (dict "resources" $component.resources) -}}
+   {{- if eq $hasResources "false" -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.eclccserver -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources := include "hpcc.checkResources" (dict "resources" $component.resources) -}}
+   {{- if eq $hasResources "false" -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.thor -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources1 := include "hpcc.checkResources" (dict "resources" $component.managerResources) -}}
+   {{- $hasResources2 := include "hpcc.checkResources" (dict "resources" $component.workerResources) -}}
+   {{- $hasResources3 := include "hpcc.checkResources" (dict "resources" $component.eclAgentResources) -}}
+   {{- if or (eq $hasResources1 "false") (or (eq $hasResources1 "false") (eq $hasResources1 "false")) -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
```

Comment:
also applies to dali, roxie etc.  Is the plan to add those in a later PR?

Reply:
Yes, they'll be checked in HPCC-28614.  This one is only to check components that affects WU costs.